src/dataset-builder/ConceptSearch.test.ts

```diff
@@ -0,0 +1,187 @@
+
+  const renderSearch = (initialSearch = '', initialCart: SnapshotBuilderConcept[] = []) =>
+    render(
+      h(ConceptSearch, {
+        actionText,
+        initialSearch,
+        initialCart,
+        onCancel,
+        onCommit,
+        onOpenHierarchy,
+        datasetId,
+        domainOption,
+      })
+    );
+
+  const displayedConcepts = [dummyGetConceptForId(102), dummyGetConceptForId(103)];
+  const mockSearch = jest.fn();
+
+  const mockDataRepoContract: DataRepoContract = {
+    dataset: (_datasetId) =>
+      ({
+        searchConcepts: mockSearch,
+      } as Partial<DataRepoContract['dataset']>),
+  } as Partial<DataRepoContract> as DataRepoContract;
+  asMockedFn(DataRepo).mockImplementation(() => mockDataRepoContract as DataRepoContract);
+  mockSearch.mockResolvedValue({
+    result: displayedConcepts,
+  });
+
+  it('renders the concept search with an initial search result', async () => {
+    // Arrange
+    renderSearch();
+    // Assert
+    expect(await screen.findByText(displayedConcepts[0].name)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[0].id)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[1].name)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[1].id)).toBeTruthy();
+    // Action text not visible until a row is selected.
+    expect(screen.queryByText(actionText)).toBeFalsy();
+  });
+
+  it('searches based on user input', async () => {
+    // Arrange
+    renderSearch();
+    const searchText = 'search text';
+    // Act
+    const user = userEvent.setup();
+    await user.type(screen.getByPlaceholderText('Search'), searchText);
+    // Assert - the search is called with the search text.
+    expect(mockSearch).toHaveBeenCalledWith(domainOption.root, searchText);
+  });
+
+  it('filters based on search text', async () => {
+    // Arrange/Act
+    const searchText = 'search text';
+    // Need to explicitly call act() here to avoid a warning about state updates during rendering.
+    await act(() => renderSearch(searchText));
+    // Assert - the initial page load should have called search with the initial search text.
+    expect(mockSearch).toHaveBeenCalledWith(domainOption.root, searchText);
+  });
+
+  it('calls open hierarchy on click', async () => {
+    // Arrange
+    const concept = displayedConcepts[0];
+    renderSearch();
+    await screen.findByText(concept.name);
+    // Act
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText(`add ${concept.id}`));
+    await user.click(screen.getByLabelText(`open hierarchy ${concept.id}`));
+    // Assert
+    expect(onOpenHierarchy).toHaveBeenCalledWith(
+      { id: concept.id, category: domainOption.category, root: concept },
+      [concept],
+      ''
+    );
+  });
+
+  it('supports add to cart', async () => {
+    // Arrange
+    renderSearch();
+    const concept = displayedConcepts[0];
+    await screen.findByText(concept.name);
```

Comment:
Nit: Do you need this line?

Reply:
Without it the test fails because the next `screen.getByText()` fails because the page hasn't rendered yet. I could instead use `find` instead of `get` to remove this from these tests, e.g.

```
    await user.click(await screen.findByLabelText(`add ${concept.id}`));
```

Reply:
I've updated the tests to `await` the first text in each test case instead of using an await at the start to wait for the page to load. Not sure which way is best but this does avoid the "unnecessary" await call after render.